File: stress/t/ctl_earlyflag.f
      PROGRAM CTLEARLY
      INTEGER*8 I,SUM,VAL,LASTIDX
      LOGICAL DONE
      SUM=0
      DONE=.FALSE.
      LASTIDX=0
      I=0
   10 IF (I.GE.50) GOTO 20
      IF (DONE) THEN
        SUM=SUM+0
      ELSE
        VAL=I*I-30
        IF (VAL.GT.100) THEN
          DONE=.TRUE.
          LASTIDX=I
        ELSE
          SUM=SUM+VAL
        END IF
      END IF
      I=I+1
      GOTO 10
   20 CONTINUE
      WRITE(*,'(A,I0)') 'sum=',SUM
      WRITE(*,'(A,I0)') 'lastidx=',LASTIDX
      END

File: stress/t/ctl_earlyflag.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t i=0,sum=0,val=0,lastidx=0;
  int done=0;
  for(i=0;i<50;i=i+1){
    if(done){ sum=sum+0; }
    else {
      val=i*i-30;
      if(val>100){ done=1; lastidx=i; }
      else { sum=sum+val; }
    }
  }
  printf("sum="); printf("%ld",(long)sum); printf("\n");
  printf("lastidx="); printf("%ld",(long)lastidx); printf("\n");
  return 0;
}
